The code comes from the patch below however in 6.12 the rk3588s.dtsi file has been changed as per 
https://github.com/torvalds/linux/commit/def88eb4d8365a4aa064d28405d03550a9d0a3be
All the code seems to be split in "-base" and "-extra"so this patch is now applied to rk3588-base.dtsi 


From d1b19bd14f99722d9d2ab9f6774427650c8ce02b Mon Sep 17 00:00:00 2001
From: Corentin Labbe <clabbe@baylibre.com>
Date: Tue, 7 Nov 2023 15:55:29 +0000
Subject: [PATCH] ARM64: dts: rk3588: add crypto node

The rk3588 has a crypto IP handled by the rk3588 crypto driver so adds a
node for it.

Signed-off-by: Corentin Labbe <clabbe@baylibre.com>
---

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
index fc67585b64b7..2efa14a2cf5c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi
@@ -1878,6 +1878,18 @@ sdhci: mmc@fe2e0000 {
 		reset-names = "core", "bus", "axi", "block", "timer";
 		status = "disabled";
 	};
+ 
+	crypto: crypto@fe370000 {
+		compatible = "rockchip,rk3588-crypto";
+		reg = <0x0 0xfe370000 0x0 0x2000>;
+		interrupts = <GIC_SPI 209 IRQ_TYPE_LEVEL_HIGH 0>;
+		clocks = <&scmi_clk SCMI_CRYPTO_CORE>, <&scmi_clk SCMI_ACLK_SECURE_NS>,
+			 <&scmi_clk SCMI_HCLK_SECURE_NS>;
+		clock-names = "core", "aclk", "hclk";
+		resets = <&scmi_reset SRST_CRYPTO_CORE>;
+		reset-names = "core";
+		status = "okay";
+	};
 
 	i2s0_8ch: i2s@fe470000 {
 		compatible = "rockchip,rk3588-i2s-tdm";

